cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -391,7 +391,7 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
     class ReconciliationState {
         private final String namespace;
         private final String name;
-        private final Kafka kafkaAssembly;
+        private Kafka kafkaAssembly;
```

Comment:
I don't think you should not do this. There is a reason why this is final. Having it not final can break many different things.